lib/mirage.ml

```diff
@@ -1566,10 +1566,22 @@ let get_mode () = Key.(get (get_base_context ()) target)
 
 let libraries_ref = ref []
 let add_to_ocamlfind_libraries l =
+  let pp_v fmt lib = Format.fprintf fmt "\"%s\" ;" lib in
+  let list_pp fmt l = Format.pp_print_list pp_v fmt l in
+  Format.eprintf "add_to_ocamlfind_libraries [%a] is deprecated. \
+      Please instead call `register` with `~libraries:[%a]`.\n"
+      list_pp l list_pp l
+  ;
```

Comment:
Recreated here: https://github.com/yomimono/mirage/tree/deprecation_annotations , specifically https://github.com/yomimono/mirage/blob/deprecation_annotations/lib/mirage_key.mli#L41 and https://github.com/yomimono/mirage/blob/deprecation_annotations/lib/mirage.mli#L438 . I get warnings when building the mirage front-end tool itself, but not when running `mirage configure` or `make`.  (I'm using a unikernel which calls both `add_to_ocamlfind_libraries` and `add_to_opam_packages`, and invoking it with `mirage configure --unix`.)